File: Ios-Swift/driver/driver/Networking/HTTP/LocationUpdate.swift
//
//  LocationUpdate.swift
//  driver
//

import Foundation

import MapKit

public class LocationUpdate: HTTPRequest {
    public var path: String = "driver/update_location"
    
    public typealias ResponseType = EmptyClass
    public var params: [String : Any]?
    
    init(jwtToken: String, location: CLLocationCoordinate2D, inTravel: Bool = false) {
        self.params = [
            "token": jwtToken,
            "location": try! location.asDictionary(),
            "inTravel": inTravel
        ]
    }
}

File: Android-Kotlin/driver/src/main/java/com/chatanoga/cab/driver/networking/http/LocationUpdate.kt
package com.chatanoga.cab.driver.networking.http

import com.google.android.gms.maps.model.LatLng
import com.chatanoga.cab.common.networking.http.interfaces.HTTPRequest

class LocationUpdate(jwtToken: String, location: LatLng, inTravel: Boolean): HTTPRequest() {
    override val path: String = "driver/update_location"
    init {
        this.params = mapOf(
                "token" to jwtToken,
                "location" to mapOf("x" to location.longitude, "y" to location.latitude),
                "inTravel" to inTravel
        )
    }
}